packages/@blockprotocol/graph/src/stdlib/subgraph/edge/link-entity.ts

```diff
@@ -0,0 +1,225 @@
+import { Entity, EntityId } from "../../../types/entity.js";
+import { OutwardEdge, Subgraph } from "../../../types/subgraph.js";
+import {
+  isHasLeftEntityEdge,
+  isHasRightEntityEdge,
+  isIncomingLinkEdge,
+  isOutgoingLinkEdge,
+  OutgoingLinkEdge,
+} from "../../../types/subgraph/edges/outward-edge-alias.js";
+import { mustBeDefined } from "../../must-be-defined.js";
+import { getEntity } from "../element/entity.js";
+
+const convertTimeToStringWithDefault = (
+  timestamp?: Date | string,
+  defaultConstructor = () => new Date().toISOString(),
+) => {
+  return timestamp === undefined
+    ? defaultConstructor()
+    : typeof timestamp === "string"
+    ? timestamp
+    : timestamp.toISOString();
+};
+
+const getUniqueEntitiesFilter = () => {
+  const set = new Set();
+  return (entity: Entity) => {
+    const editionIdString = JSON.stringify(entity.metadata.editionId);
+    if (set.has(editionIdString)) {
+      return false;
+    } else {
+      set.add(editionIdString);
+      return true;
+    }
+  };
+};
+
+/** @todo - Update these methods to take intervals instead of timestamps */
+
+/**
+ * For a given moment in time, get all outgoing link entities from a given entity.
+ *
+ * @param subgraph
+ * @param {EntityId} entityId - The ID of the source entity to search for outgoing links from
+ * @param {Date | string} [timestamp] - An optional `Date` or an ISO-formatted datetime string of the moment to search
+ *    for, if not supplied it defaults to the current time
+ */
+export const getOutgoingLinksForEntity = (
+  subgraph: Subgraph,
+  entityId: EntityId,
+  timestamp?: Date | string,
+): Entity[] => {
+  const timestampString = convertTimeToStringWithDefault(timestamp);
+
+  const entityEdges = subgraph.edges[entityId];
+
+  if (!entityEdges) {
+    return [];
+  }
+
+  const uniqueEntitiesFilter = getUniqueEntitiesFilter();
+
+  return (
+    Object.entries(entityEdges)
+      // Only look at outgoing edges that were created before or at the timestamp
+      .filter(([edgeTimestamp, _]) => edgeTimestamp <= timestampString)
+      // Extract the link `EntityEditionId`s from the endpoints of the link edges
+      .flatMap(([_, outwardEdges]) => {
+        return (outwardEdges as OutwardEdge[])
+          .filter(isOutgoingLinkEdge)
+          .map((edge) => {
+            return edge.rightEndpoint;
+          });
+      })
+      .map(({ baseId: linkEntityId, timestamp: _firstEditionTimestamp }) => {
+        return mustBeDefined(
+          getEntity(
+            subgraph,
+            linkEntityId,
+            // Find the edition of the link at the given moment (not at `_firstEditionTimestamp`, the start of its history)
+            timestampString,
+          ),
+        );
+      })
+      .filter((x): x is Entity => x !== undefined)
```

Comment:
It would, I noticed that after reading your previous comment, it's removed 
@Alfred-Mountfield
[Remove unnecessary undefined checks](https://github.com/blockprotocol/blockprotocol/pull/820/commits/ebab17cf139e04667baa4705fb72afa07e6858fa)